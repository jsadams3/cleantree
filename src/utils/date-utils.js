export const formatDateString = (ISODateString) => {
  const options = { month: 'long', day: 'numeric', year: 'numeric' }
  const date = new Date(ISODateString)
  return new Intl.DateTimeFormat('en-US', options).format(date)
}
