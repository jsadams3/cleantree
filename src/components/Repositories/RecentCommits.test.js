import React from 'react'
import { shallow } from 'enzyme'
import { RecentCommits } from './'
import { formatDateString } from '../../utils'

const props = {
  author: {
    avatarUrl: 'http://via.placeholder.com/640x360',
    name: 'Bronwyn',
  },
  committedDate: '2020-11-09T01:01:59.975Z',
  message: '',
}
describe('RecentCommits', () => {
  it('should render the component', () => {
    const component = shallow(<RecentCommits node={props} />)

    expect(component).toMatchSnapshot()
  })

  it('should take an ISO-8601 encoded UTC date string and format it correctly', () => {
    const ISODateString = '2020-11-10T16:20:13.750Z'
    expect(formatDateString(ISODateString)).toEqual('November 10, 2020')
  })
})
