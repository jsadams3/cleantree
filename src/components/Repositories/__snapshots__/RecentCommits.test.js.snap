// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentCommits should render the component 1`] = `
<div
  className="commit-container"
>
  <img
    alt="Bronwyn"
    className="commit-author-avatar"
    src="http://via.placeholder.com/640x360"
  />
  <div
    className="commit"
  >
    <p
      className="commit-message"
    />
    <span
      className="commit-info"
    >
      <p
        className="commit-author-name"
      >
        Bronwyn
      </p>
      <span>
         committed on November 8, 2020
      </span>
    </span>
  </div>
</div>
`;
